x-pack/plugin/esql/qa/testFixtures/src/main/resources/ints.csv-spec

```diff
@@ -986,3 +986,22 @@ s:double | emp_no:integer | salary:integer
 1.0     | 10002          | 56371
 1.0     | 10041          | 56415
 ;
+
+CombineBinaryComparisons
+required_capability: combine_binary_comparisons
+
+from employees
+| where salary_change.int == 2 OR salary_change.int > 1
+| keep emp_no, salary_change.int
+| sort emp_no
+;
+warning:Line 2:35: evaluation of [salary_change.int > 1] failed, treating result as null. Only first 20 failures recorded.
+warning:Line 2:35: java.lang.IllegalArgumentException: single-value function encountered multi-value
+
+emp_no:integer |salary_change.int:integer
+10044 | 8
+10046 | 2
+10066 | 5
+10079 | 7
+10086 | 13
+;
```

Comment:
Please, add this test as well `row x = [1,2,3] | where 12 * (-x - 5) >= -120 OR x < 5`. It's the same formula that this PR tries to fix.

Reply:
Will add it.